main.nf

```diff
@@ -60,6 +93,52 @@ logs_names = [
 "drawTree": "rate_adjustment.log" ]
 
 
+/*
+ * Function to log various process information before submission/execution
+ * if print is set to true or if any process setting deviates from defaults
+ * (executor = 'local', cpus = 1, and no memory setting).
+ */
+def logProcessInfo(task, speciesLabel="", printIndex=false, print=LOG_PROCESS_INFO) {
+    if (print || task.executor != 'local' || task.cpus != 1 || task.memory) {
+        log.info "[${task.process}]${ speciesLabel ? ' [' + speciesLabel + '] ' : ' ' }" +\
+                 "Process${ printIndex ? ' (' + task.index + ') ' : ' ' }" +\
+                 "queued on executor '${task.executor}' with " +\
+                 "${task.cpus} ${ task.cpus == 1 ? 'CPU' : 'CPUs' }" +\
+                 "${ task.memory ? ' and ' + task.memory : '' }"
```

Comment:
I noticed that if the user sets the memory requirement through `clusterOptions` instead of `memory`, there will be no memory shown in this log line.
Perhaps we could add also clusterOptions options at the end of this log.info? Not really urgent, but to complete the purpose of this logger.

Reply:
Yeah, we could and I thought about it, but decided to only use the most basic and common ones. `clusterOptions` is a pretty specific one, and then we should also think about adding `time`, `queue` etc. and then it gets too long.

Reply:
Makes sense :)